### PR TITLE
Cache spec output

### PR DESCRIPTION
**Pull request by illicitonion**
Does not cache inability to generate spec. Does not cache diffs or
anything fancy. Just the raw spec generation.

### CODE REVIEW

File: scripts/speculator/main.go [hunk 1]
```diff
@@ -387,3 +398,9 @@ func serveText(s string) func(http.ResponseWriter, *http.Request) {
 		io.WriteString(w, s)
 	}
 }
+
+func initCache() error {
+	c, err := lru.New(50)
```
Comment (Kegsay) [inline scripts/speculator/main.go]: `50` what? \*picks random unit\* ... [Ponies](https://en.wikipedia.org/wiki/List_of_unusual_units_of_measurement#Pony)

File: scripts/speculator/main.go [hunk 2]
```diff
@@ -25,6 +25,8 @@ import (
 	"strings"
 	"syscall"
 	"time"
+
+	"github.com/hashicorp/golang-lru"
```
Comment (Kegsay) [inline scripts/speculator/main.go]: I don't know if Go does this but what if you know, the library gets updated in a backwards-incompatible way? Can we pin a version on this?

**Comment by Kegsay** [comment]
LGTM other than comments.

File: scripts/speculator/main.go (hunk 1 above)
Comment (illicitonion) [inline scripts/speculator/main.go]: Ponies indeed.

Added comment.

File: scripts/speculator/main.go (hunk 2 above)
Comment (illicitonion) [inline scripts/speculator/main.go]: Nope; there are a couple of package managers that have evolved, and in the most recent release, the core team have put together a "we want to solve this properly" experimental solution, but it's not a thing yet: https://docs.google.com/document/d/1Bz5-UB7g2uPBdOx-rw5t9MxJwkfpx90cqG9AFL0JAYo/edit

**Comment by illicitonion** [comment]
Manually merged

**Pull request closed without merging**